<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Day - 4 Functional Analysis, Longitudinal &amp; Public Data | r4proteomics</title>
<meta name="author" content="Miguel CASANOVA, PhD &amp; Dany MUKESHA">
<meta name="description" content="4.1 Objectives Interpret DE results biologically Explore longitudinal trajectories Work with public datasets  4.2 Module 1: Functional Enrichment &amp; GSEA Introduce enrichment (GO, KEGG, Reactome)....">
<meta name="generator" content="bookdown 0.44 with bs4_book()">
<meta property="og:title" content="Day - 4 Functional Analysis, Longitudinal &amp; Public Data | r4proteomics">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/lodiag-holocron/r4proteomics/day4.html">
<meta property="og:description" content="4.1 Objectives Interpret DE results biologically Explore longitudinal trajectories Work with public datasets  4.2 Module 1: Functional Enrichment &amp; GSEA Introduce enrichment (GO, KEGG, Reactome)....">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Day - 4 Functional Analysis, Longitudinal &amp; Public Data | r4proteomics">
<meta name="twitter:description" content="4.1 Objectives Interpret DE results biologically Explore longitudinal trajectories Work with public datasets  4.2 Module 1: Functional Enrichment &amp; GSEA Introduce enrichment (GO, KEGG, Reactome)....">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="5-Day Training Course">r4proteomics</a>:
        <small class="text-muted">5-Day Training Course</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome</a></li>
<li><a class="" href="day1.html"><span class="header-section-number">1</span> Introduction to R and RStudio</a></li>
<li><a class="" href="day2.html"><span class="header-section-number">2</span> Introduction to Proteomic Data &amp; Quality Control</a></li>
<li><a class="" href="day3.html"><span class="header-section-number">3</span> Preprocessing and Differential Expression</a></li>
<li><a class="active" href="day4.html"><span class="header-section-number">4</span> Functional Analysis, Longitudinal &amp; Public Data</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/lodiag-holocron/r4proteomics">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="day4" class="section level1" number="4">
<h1>
<span class="header-section-number">Day - 4</span> Functional Analysis, Longitudinal &amp; Public Data<a class="anchor" aria-label="anchor" href="#day4"><i class="fas fa-link"></i></a>
</h1>
<div id="objectives" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Objectives<a class="anchor" aria-label="anchor" href="#objectives"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Interpret DE results biologically<br>
</li>
<li>Explore longitudinal trajectories<br>
</li>
<li>Work with public datasets</li>
</ul>
</div>
<div id="module-1-functional-enrichment-gsea" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Module 1: Functional Enrichment &amp; GSEA<a class="anchor" aria-label="anchor" href="#module-1-functional-enrichment-gsea"><i class="fas fa-link"></i></a>
</h2>
<p>Introduce enrichment (GO, KEGG, Reactome).<br>
GSEA principles and visualization.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example with clusterProfiler</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="co"># Suppose `de_genes` is a vector of gene IDs with statistics</span></span>
<span><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span><span class="va">de_genes</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, keyType <span class="op">=</span> <span class="st">"ENSEMBL"</span>, ont <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">ego</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lme4/lme4/">lme4</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: Matrix</span></span>
<span><span class="va">df_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>  sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,</span>
<span>  time <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, times <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>  intensity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">intensity</span> <span class="op">~</span> <span class="va">time</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="va">time</span> <span class="op">|</span> <span class="va">sample</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">df_long</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span>
<span><span class="co">#&gt; Linear mixed model fit by REML ['lmerMod']</span></span>
<span><span class="co">#&gt; Formula: intensity ~ time + (1 + time | sample)</span></span>
<span><span class="co">#&gt;    Data: df_long</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; REML criterion at convergence: 65</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Scaled residuals: </span></span>
<span><span class="co">#&gt;     Min      1Q  Median      3Q     Max </span></span>
<span><span class="co">#&gt; -1.2362 -0.4703 -0.1740  0.6994  1.5685 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Random effects:</span></span>
<span><span class="co">#&gt;  Groups   Name        Variance Std.Dev. Corr </span></span>
<span><span class="co">#&gt;  sample   (Intercept) 0.16323  0.4040        </span></span>
<span><span class="co">#&gt;           time        0.02183  0.1477   -0.12</span></span>
<span><span class="co">#&gt;  Residual             0.34059  0.5836        </span></span>
<span><span class="co">#&gt; Number of obs: 30, groups:  sample, 10</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Fixed effects:</span></span>
<span><span class="co">#&gt;             Estimate Std. Error t value</span></span>
<span><span class="co">#&gt; (Intercept)   0.6365     0.2114   3.010</span></span>
<span><span class="co">#&gt; time         -0.5349     0.1386  -3.859</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Correlation of Fixed Effects:</span></span>
<span><span class="co">#&gt;      (Intr)</span></span>
<span><span class="co">#&gt; time -0.605</span></span></code></pre></div>
<div id="exercise" class="section level3" number="4.2.1">
<h3>
<span class="header-section-number">4.2.1</span> Exercise<a class="anchor" aria-label="anchor" href="#exercise"><i class="fas fa-link"></i></a>
</h3>
<p>On a longitudinal proteomics dataset, plot trajectories per protein or per cluster, fit simple linear/mixed model.</p>
</div>
</div>
<div id="module-2-public-data-integration" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Module 2: Public Data Integration<a class="anchor" aria-label="anchor" href="#module-2-public-data-integration"><i class="fas fa-link"></i></a>
</h2>
<p>Demonstrate downloading, cleaning, integrating a public proteomics or expression dataset (e.g. from PRIDE, GEO). Run pipeline: QC → normalization → DE → enrichment.</p>
<div id="exercise-1" class="section level3" number="4.3.1">
<h3>
<span class="header-section-number">4.3.1</span> Exercise<a class="anchor" aria-label="anchor" href="#exercise-1"><i class="fas fa-link"></i></a>
</h3>
<p>Pick a public dataset, import into R, preprocess, analyze, interpret results.</p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="day3.html"><span class="header-section-number">3</span> Preprocessing and Differential Expression</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#day4"><span class="header-section-number">4</span> Functional Analysis, Longitudinal &amp; Public Data</a></li>
<li><a class="nav-link" href="#objectives"><span class="header-section-number">4.1</span> Objectives</a></li>
<li>
<a class="nav-link" href="#module-1-functional-enrichment-gsea"><span class="header-section-number">4.2</span> Module 1: Functional Enrichment &amp; GSEA</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#exercise"><span class="header-section-number">4.2.1</span> Exercise</a></li></ul>
</li>
<li>
<a class="nav-link" href="#module-2-public-data-integration"><span class="header-section-number">4.3</span> Module 2: Public Data Integration</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#exercise-1"><span class="header-section-number">4.3.1</span> Exercise</a></li></ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/lodiag-holocron/r4proteomics/blob/master/day4_functional_public.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/lodiag-holocron/r4proteomics/edit/master/day4_functional_public.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>r4proteomics</strong>: 5-Day Training Course" was written by Miguel CASANOVA, PhD &amp; Dany MUKESHA. It was last built on 2025-10-02.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
